### DISCUSSION TITLE
OCI repoURL for Helm charts in ArgoCD gets picked up as docker image updates

### DISCUSSION THREAD

**Discussion by p4block** [renovatebot/renovate#33143] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

Bitnami deprecated https artifacts, so we moved to oci endpoints for the chart repos. 

We have apps declared as Application custom resources from ArgoCD and those now create a dependency named "repoURL+chart"

Example with `influxdb-release.yaml`
```
  source:
    repoURL: registry-1.docker.io/bitnamicharts
    targetRevision: 6.3.23
    chart: influxdb 
```

Now creates an MR to update `registry-1.docker.io/bitnamicharts/influxdb`  and broke our custom rules on auto upgrading (that were targeting just "influxdb")

This didn't happen before when `repoURL: "https://charts.bitnami.com/bitnami"`

I have some ideas on how to fix it, but wonder what would be the correct solution to work around the problem. I think that some config:recommended rule was being at play but is no longer acting.

**Comment by rarkins** [comment]
Could you try to edit the above for more clarity? I'm not sure what you're asking.

**Reply by p4block** [reply]
Let me rephrase
We switched repoURL from "https://charts.bitnami.com/bitnami" (https) to registry-1.docker.io/bitnamicharts (oci) and now package names are detected as registry-1.docker.io/bitnamicharts/package-name instead of just package-name

PR names are like 
` chore(deps): update registry-1.docker.io/bitnamicharts/influxdb docker tag to v6.5.0 `

Which now come think of it, **it's treating them now as docker images and not helm charts.**
`chore(deps): update helm release sonarqube to v10.8.0`

If I go into Dependecy Dashboard I see they're all under the ArgoCD category in detected dependencies though.

**Reply by p4block** [reply]
Note that I don't particularly mind that, I just need the detected package name to not include the repo base url. I could do it with a metric ton of custom package groups I think, but something tells me that's not the best approach.

**Reply by rarkins** [reply]
It's correct that `packageName` is fully qualified, e.g. like `registry-1.docker.io/bitnamicharts/influxdb` above. What you see in the PR title is based on `depName`, which is possible to make different than `packageName`. I guess in this case it could make sense to call that `bitnamicharts/influxdb`.

**Reply by p4block** [reply]
I think there's a bigger problem than aesthetics, charts used to be detected as charts with the old https endpoint but with the oci endpoint the chart is detected as a "docker image update" (and I also get no changelog of the chart)

Iirc it should still be possible to obtain this info from the oci registry, but in any case, they're not docker images (!)

**Reply by rarkins** [reply]
The fact it says "docker image" _is_ aesthetics. It's from this default config from the docker datasource: https://github.com/renovatebot/renovate/blob/fc36735f0c7dcf174122b455ab499d6d41e34127/lib/modules/datasource/docker/index.ts#L52

You could probably override this with packageRules and setting `commitMessageTopic`.

We could perhaps set up one or more new datasources which are essentially "wrappers" around datasource/docker but which have different config templates.

**Reply by p4block** [reply]
I've let it do its thing for two days and now I have a clear picture of what happened.
I checked more and you're right, the naming is aesthetics and changelog still works for some charts even if it thinks it's docker images. I've just renamed it in our configuration, thanks for the heads up. 

Back to the original title of the discussion, what _is_ causing mayhem is the inconsistent dependency names when using OCI endpoints.  

ArgoCD CRDs get the full path to the chart as names but helmchart dependencies just the chart name, so we get duplicate PRs for services that we deploy both ways. 

ArgoCD resources get PRs like `Update registry-1.docker.io/bitnamicharts/mysql Docker tag to v12.2.0`

While helmchart dependencies are just the chart name without the path
The following chart causes a PR to be created under name  `Update mysql Docker tag to v12.2.0`
```
apiVersion: v2
name: blog
version: 1.0.0
dependencies:
- name: wordpress
  version: 24.1.4
  repository: oci://registry-1.docker.io/bitnamicharts
- name: mysql
  version: 12.1.0
  repository: oci://registry-1.docker.io/bitnamicharts

```

No matter how a helmchart (or docker image) is detected, the PRs should follow the same naming convention and structure. It was just "mysql" before but "registry-1.docker.io/bitnamicharts/mysql" is also fine. The problem is that it's suddenly doing _both_ after changing a URL. 
That part is not aesthetics because it requires a heavy rework of our renovate configuration and broke things very suddenly for what would seem to be an innocuous change. 

Should I open another discussion for that?

**Reply by rarkins** [reply, marked as answer]
Sure, let's wrap this original post up with: Renovate uses the "docker" datasource for anything on docker.io and by default these are labeled as "docker tag x" even if not technically docker. It's possible to override this via config.

For the topic of "should Renovate use a string like registry-1.docker.io/bitnamicharts/mysql or trim it to bitnamicharts/mysql or mysql?" then please start a separate discussion. I'm not sure I agree with your assumption that Renovate can/should be opinionated enough to use "mysql" or that it should be responsible for overcoming a change in how bitnami publishes.
